### ISSUE TITLE
TypeGuards isSemicolonToken, isInferKeyword negated will infer node as 'never'

### ISSUE THREAD

**Issue by cancerberoSgx** [dsherret/ts-morph#728]
This two typeguards, isSemicolonToken, isInferKeyword , when used with negation logic operator will infer the node has type 'never' which I think is not correct, as shown :

<img width="444" alt="Screen Shot 2019-10-05 at 3 05 46 PM" src="https://user-images.githubusercontent.com/112746/66258968-c72e2b80-e781-11e9-848e-d12a4f97f07e.png">

snippet : 

```ts
import { Project, TypeGuards } from 'ts-morph'

const project = new Project()
const file = project.createSourceFile('f1.ts', 'var a = 1')
var n = file.getDescendants()[2]
if (n && !TypeGuards.isSemicolonToken(n)) {
  var c = n.getKindName()
}
```

I think this could be a solution but I wanted to ask before PR:

```ts
    static readonly isSemicolonToken: (node: Node) => node is Node<ts.Token<SyntaxKind.SemicolonToken>>;
```

**Comment by dsherret** [comment]
@cancerberoSgx oops! Yup, it should be that. Feel free to open a PR.

**Comment by dsherret** [comment]
I fixed this along with #760. The fix was a bit more involved for this one than I thought. I really need to clean up the code generation code...
